core/util/runPuppet.js

```diff
@@ -71,114 +74,116 @@ async function processScenarioView (scenario, variantOrScenarioLabelSafe, scenar
   if (chromeVersion < MIN_CHROME_VERSION) {
     console.warn(`***WARNING! CHROME VERSION ${MIN_CHROME_VERSION} OR GREATER IS REQUIRED. PLEASE UPDATE YOUR CHROME APP!***`);
   }
-
-  // --- BEFORE SCRIPT ---
-  var onBeforeScript = scenario.onBeforeScript || config.onBeforeScript;
-  if (onBeforeScript) {
-    var beforeScriptPath = path.resolve(engineScriptsPath, onBeforeScript);
-    if (fs.existsSync(beforeScriptPath)) {
-      await require(beforeScriptPath)(page, scenario, viewport, isReference, browser);
-    } else {
-      console.warn('WARNING: script not found: ' + beforeScriptPath);
+  
+  let result;
+  const puppetCommands = async () => {
+    // --- BEFORE SCRIPT ---
+    var onBeforeScript = scenario.onBeforeScript || config.onBeforeScript;
+    if (onBeforeScript) {
+      var beforeScriptPath = path.resolve(engineScriptsPath, onBeforeScript);
+      if (fs.existsSync(beforeScriptPath)) {
+        await require(beforeScriptPath)(page, scenario, viewport, isReference, browser);
+      } else {
+        console.warn('WARNING: script not found: ' + beforeScriptPath);
+      }
     }
-  }
 
-  //  --- OPEN URL ---
-  var url = translateUrl(scenario.url);
-  if (isReference && scenario.referenceUrl) {
-    url = scenario.referenceUrl;
-  }
-  await page.goto(url);
-
-  await injectBackstopTools(page);
-
-  //  --- WAIT FOR READY EVENT ---
-  var readyEvent = scenario.readyEvent || config.readyEvent;
-  if (readyEvent) {
-    await page.evaluate(`window._readyEvent = '${readyEvent}'`)
+    //  --- OPEN URL ---
+    var url = translateUrl(scenario.url);
+    if (isReference && scenario.referenceUrl) {
+      url = scenario.referenceUrl;
```

Comment:
Holy crap -- I just found a bug that I introduced a few days ago!   This value should also be wrapped in a `translateUrl `   -- would you mind adding that?

`url = translateUrl(scenario.referenceUrl);`

Reply:
done!